include/boost/unordered/detail/implementation.hpp

```diff
@@ -1913,6 +1917,44 @@ namespace boost {
           }
         }
 
+      private:
+        class allocator_constructed_key_type
+        {
+          using key_type = typename Types::key_type;
+          using key_allocator_type =
+            typename detail::rebind_wrap<node_allocator_type, key_type>::type;
+
+          alignas(key_type) unsigned char storage[sizeof(key_type)];
```

Comment:
We already have this pattern in the codebase, check out https://github.com/boostorg/unordered/blob/67c5cdb3a69f0b92d2779880ce9aa1d46e54cf7b/include/boost/unordered/detail/opt_storage.hpp

Reply:
I tried using `opt_storage` here and it's fine, but it gives a compile error because of deleted copy constructor in the `detail::foa::allocator_constructed` class created for this exact same purpose. I think instead I'll factor these 2 classes into a single implementation